fastlane_core/lib/fastlane_core/fastlane_pty.rb

```diff
@@ -31,6 +40,11 @@ def self.spawn(command)
         command_stdout.close
         p.value.exitstatus
       end
+    rescue StandardError => e
+      # Wrapping any error in FastlanePtyError to allow
+      # callers to see and use $?.exitstatus that
+      # would usually get returned
+      raise FastlanePtyError.new(e, $?.exitstatus)
```

Comment:
Is `$?.exitstatus` actually set here? Above it uses and returns `p.value.exitstatus`...

Reply:
Ah, I guess this only supports the exit status with `PTY.spawn` but not with `Open3` 🤔 I can do a follow up PR

Reply:
I am not sure how Open3 bubbles up the exit status... will have to look into it - and best create a test to confirm ;)